src/Pulsar.Client/Schema/AvroSchema.fs

```diff
@@ -84,8 +87,11 @@ type internal GenericAvroSchema(topicSchema: TopicSchema) =
             schemaFields
             |> Seq.map (fun sf -> { Name = sf.Name; Value = record.[sf.Name]; Index = sf.Pos })
             |> Seq.toArray
-        let scemaVersionBytes =
-            topicSchema.SchemaVersion
+        let schemaVersionBytes =
+            schemaVersion
             |> Option.map (fun sv -> sv.Bytes)
             |> Option.toObj
-        GenericRecord(scemaVersionBytes, fields)
+        GenericRecord(schemaVersionBytes, fields)
+        
+    override this.GetSpecificSchema schemaInfo schemaVersion =
```

Comment:
Can you please make it a tuple (since ISchema is public and C# doesn't support curried parameters well)